### PR TITLE
Implement integration test

### PR DESCRIPTION
**Pull request by lordksix**
In this PR, the changes are as follows: 
  - [x] Make sure that the N+1 problem is solved when fetching all posts and their comments for a user.
  previous
Screenshot from 2023-08-29 16-09-40
  new
Screenshot from 2023-08-29 16-08-51

  - [x] Implement integration test for user index page
  - [x] Implement integration test for user show page
  - [x] Implement integration test for user post index page:
  - [x] Implement integration test for post show page:

### CODE REVIEW

File: spec/views/users/user_index_spec.rb [hunk 1]
```diff
@@ -3,17 +3,17 @@
...
     @user = User.create(
-        name: 'Max',
-        bio: 'Full-stack developer',
-        photo: 'http://localhost/123',
-        post_counter: 0
+      name: 'Max',
+      bio: 'Full-stack developer',
+      photo: 'http://localhost/123',
+      post_counter: 0
     )
```
Comment (abdulhamiid) [inline spec/views/users/user_index_spec.rb]: According to the project requirements, please use Capybara to write integration tests for the user show page: 🤗🛠

image

I suggest that you implement that in a separate file so as to make sure that individual parts of your application work. 💪

File: spec/views/posts/post.show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      visit user_post_path(@user.id, @post_one.id)
+      expect(page).to have_content('Max')
+    end
+
+    it 'Can see the comment each commentor left.' do
+      visit user_post_path(@user.id, @post_one.id)
+      expect(page).to have_content('Comment1')
+    end
+  end
+end
```
Comment (abdulhamiid) [inline spec/views/posts/post.show_spec.rb]: According to the project requirements, you have written an integration test for the post-show page 👍. 

Please write integration tests for the post index page also: 🤗🛠

image

- I suggest that you implement that in a separate file. 💪

- here's a [capybara cheat sheet](https://gist.github.com/zhengjia/428105) you can use. 🤗

**Review by abdulhamiid — CHANGES_REQUESTED** [review]
# Status: Required Changes :recycle:
Hi Team 👋, 
Alt Text
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights 🌟

- ✔️ Descriptions are very descriptive
- ✔️ Write integration tests with Capybara gem.
- ✔️ Secure app from n+1 problems.

## Required Changes ♻️
_Please kindly check the comments under the review._

## Optional suggestions 🤷‍♂️
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/views/users/user_index_spec.rb (hunk 1 above)
Comment (lordksix) [reply spec/views/users/user_index_spec.rb]: Hello, @abdulhamiid.  The tests for user index page were transfer to spec/view/user/index.html.erb_spec.rb so this is an old and incomplet version of that file. It doest not exist in the final PR.

File: spec/views/posts/post.show_spec.rb (hunk 2 above)
Comment (lordksix) [reply spec/views/posts/post.show_spec.rb]: Hello, @abdulhamiid. The tests for post show page were transfer to spec/view/post/show.html.erb_spec.rb so this is an old and incomplet version of that file. It doest not exist in the final PR.

**Review by anagudelogu — APPROVED** [review]
# STATUS: APPROVED! 🍾 🎆 🥇 

Hi @lordksix, 
I'm Andres.

You did a great job working on this milestone, your project is complete! 🚀 
There is nothing else to say other than... it's time to merge it 🤖  
Congratulations! 🎉 



## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Descriptive PR title, summary and commit messages. 💬 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**